src/main/java/com/mraof/minestuck/block/HorseClockBlock.java

```diff
@@ -0,0 +1,141 @@
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.EntityBlock;
+import net.minecraft.world.level.block.RenderShape;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.entity.BlockEntityTicker;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+import net.minecraft.world.level.block.state.properties.IntegerProperty;
+import net.minecraft.world.phys.BlockHitResult;
+
+import javax.annotation.Nullable;
+import java.util.Random;
+
+public class HorseClockBlock extends MultiMachineBlock implements EntityBlock
+{
+	public static final IntegerProperty POWER = BlockStateProperties.POWER;
+	
+	public HorseClockBlock(MachineMultiblock machine, Properties properties)
+	{
+		super(machine, properties);
+		registerDefaultState(stateDefinition.any().setValue(POWER, 0));
+	}
+	
+	@Override
+	public InteractionResult use(BlockState state, Level level, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit)
+	{
+		//will only work on the bottom block
+		if(level.getBlockEntity(pos) instanceof HorseClockBlockEntity be)
+		{
+			HorseClockBlockEntity.fullPower(level, be);
+			
+			return InteractionResult.SUCCESS;
+		}
+		
+		return InteractionResult.PASS;
+	}
+	
+	@Nullable
+	@Override
+	public BlockEntity newBlockEntity(BlockPos pos, BlockState state)
+	{
+		if(state.is(MSBlocks.HORSE_CLOCK.BOTTOM.get())) //prevents non block entity blocks from trying to tick
+		{
+			return new HorseClockBlockEntity(pos, state);
+		} else
+			return null;
+	}
+	
+	public RenderShape getRenderShape(BlockState state)
+	{
+		return RenderShape.INVISIBLE;
+	}
+	
+	@Override
+	public void neighborChanged(BlockState state, Level level, BlockPos pos, Block block, BlockPos fromPos, boolean isMoving)
+	{
+		updateSurvival(state, level, pos);
+	}
+	
+	public static void updateSurvival(BlockState state, Level level, BlockPos pos)
+	{
+		if(!level.isClientSide() && state.getBlock() instanceof HorseClockBlock)
+		{
+			if(state.is(MSBlocks.HORSE_CLOCK.BOTTOM.get()) &&
+					!level.getBlockState(pos.above()).is(MSBlocks.HORSE_CLOCK.CENTER.get()))
+				level.destroyBlock(pos, true);
+			else if(state.is(MSBlocks.HORSE_CLOCK.CENTER.get()) &&
+					(!level.getBlockState(pos.above()).is(MSBlocks.HORSE_CLOCK.TOP.get()) || !level.getBlockState(pos.below()).is(MSBlocks.HORSE_CLOCK.BOTTOM.get())))
+				level.destroyBlock(pos, true);
+			else if(state.is(MSBlocks.HORSE_CLOCK.TOP.get()) &&
+					!level.getBlockState(pos.below()).is(MSBlocks.HORSE_CLOCK.CENTER.get()))
+				level.destroyBlock(pos, true);
```

Comment:
It would've been nice to use a more general function from the multiblock to handle these validity checks, though I'm not sure if there's something that's readily available for this.

Reply:
I thought we might have had it, and looked briefly, but didnt find anything. I would have added it but did not feel inspired